lms/djangoapps/dashboard/sysadmin_get_course_users.py

```diff
@@ -0,0 +1,266 @@
+import json
+
+from django.contrib.auth.models import User
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from django.utils.translation import ugettext as _
+
+from courseware.courses import get_course_by_id
+from openedx.core.djangoapps.content.course_structures.models import CourseStructure
+from student.views import get_course_enrollments
+from student.models import CourseEnrollment, UserProfile
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from courseware import models
+from ccx.utils import get_course_chapters
+
+
+"""
+This module creates csv report with list of course users.
+"""
+
+
+def _find_course(course_key, courses):
+    if course_key in courses:
+        course = courses[course_key]
+        return course
+    else:
+        try:
+            course = get_course_by_id(course_key)
+            return course
+        except Exception:   # pylint: disable=broad-except
+            return None
+
+
+def _get_children(parent, course_ordered):
+    """
+    Method for getting children from course structure object
+
+    parent - usage key of course structure element (e.g. module_state_key), for example:
+    u'block-v1:edX+DemoX+Demo_Course+type@vertical+block@4e592689563243c484af947465eaef0d'
+    """
+    children = course_ordered[parent].get('children', [])
+    return children
+
+
+def _get_children_from_list(parents_list, course_ordered):
+    """
+    Method for getting children from course structure objects list
+
+    parent_list - list of usage keys of course structure elements (e.g. module_state_key s), for example:
+    [u'block-v1:edX+DemoX+Demo_Course+type@vertical+block@4e592689563243c484af947465eaef0d']
+    """
+    children_list = []
+    for parent in parents_list:
+        one_parent_children = _get_children(parent, course_ordered)
+        for one_parent_child in one_parent_children:
+            children_list.append(one_parent_child)
```

Comment:
you do not have to iterate through the one_parent_children, you can just add all of them to the children_list using extend()